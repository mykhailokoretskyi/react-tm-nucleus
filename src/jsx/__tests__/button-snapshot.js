import React from 'react';
import Button from '../button';
import ButtonCTA from '../buttonCTA';
import renderer from 'react-test-renderer';

describe('Button', () => {
    it(' "a" matches snapshots', () => {
        const component = renderer.create(
            <Button>
                Test button
            </Button>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(' "button" matches snapshot', () => {
        const component = renderer.create(
            <Button button={true}>
                Test button
            </Button>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(' disabled matches snapshots', () => {
        const component = renderer.create(
            <Button disabled={true}>
                Test button
            </Button>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('ButtonCTA', () => {
    it(' "a" matches snapshots', () => {
        const component = renderer.create(
            <ButtonCTA>
                Test button
            </ButtonCTA>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(' "button" matches snapshot', () => {
        const component = renderer.create(
            <ButtonCTA button={true}>
                Test button
            </ButtonCTA>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(' disabled matches snapshots', () => {
        const component = renderer.create(
            <ButtonCTA disabled={true}>
                Test button
            </ButtonCTA>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

