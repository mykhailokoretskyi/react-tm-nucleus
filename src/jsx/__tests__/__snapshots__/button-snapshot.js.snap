exports[`Button  "a" matches snapshots 1`] = `
<a
  className="button"
  href="#"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Test button
</a>
`;

exports[`Button  "button" matches snapshot 1`] = `
<button
  className="button"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Test button
</button>
`;

exports[`Button  disabled matches snapshots 1`] = `
<a
  className="button"
  disabled={true}
  href="#"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Test button
</a>
`;

exports[`ButtonCTA  "a" matches snapshots 1`] = `
<a
  className="button button--cta"
  href="#"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Test button
</a>
`;

exports[`ButtonCTA  "button" matches snapshot 1`] = `
<button
  className="button button--cta"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Test button
</button>
`;

exports[`ButtonCTA  disabled matches snapshots 1`] = `
<a
  className="button button--cta"
  disabled={true}
  href="#"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Test button
</a>
`;

exports[`ButtonFull  "a" matches snapshots 1`] = `
<a
  className="button button--full"
  href="#"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Test button
</a>
`;

exports[`ButtonFull  "button" matches snapshot 1`] = `
<button
  className="button button--full"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Test button
</button>
`;

exports[`ButtonFull  disabled matches snapshots 1`] = `
<a
  className="button button--full"
  disabled={true}
  href="#"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Test button
</a>
`;

exports[`ButtonStatic  "a" matches snapshots 1`] = `
<a
  className="button button--static"
  href="#"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Test button
</a>
`;

exports[`ButtonStatic  "button" matches snapshot 1`] = `
<button
  className="button button--static"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Test button
</button>
`;

exports[`ButtonStatic  disabled matches snapshots 1`] = `
<a
  className="button button--static"
  disabled={true}
  href="#"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Test button
</a>
`;
